### PR TITLE
Add standard labels to DLCs

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Add com.amazonaws.sagemaker.dlc labels to containers
- Currently, this is added to all DLCs - even E3 - we may want to change the name to remove the "sagemaker" prefix as a result, to avoid confusion with E3 customers.
- Add tests to ensure that labels are added
- Add new HF buildspec argument for "contributor"

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/image_builder.py [hunk 1]
```diff
@@ -139,6 +139,16 @@ def image_builder(buildspec):
         if "labels" in image_config:
             labels.update(image_config.get("labels"))
 
+        # Adding standard labels to all images
+        labels[f"com.amazonaws.sagemaker.dlc.framework.{str(BUILDSPEC['framework'])}"] = True
```
Comment (arjkesh) [inline src/image_builder.py:143]: Add job_type as well (training/inference)

File: src/image_builder.py [hunk 2]
```diff
@@ -139,6 +139,27 @@ def image_builder(buildspec):
         if "labels" in image_config:
             labels.update(image_config.get("labels"))
 
+        # Adding standard labels to all images
+        labels[f"com.amazonaws.sagemaker.dlc.framework.{str(BUILDSPEC['framework'])}"] = "true"
```
Comment (arjkesh) [inline src/image_builder.py:143]: thinking about naming this `com.amazonaws.dlc.sagemaker` - this way we can also have `com.amazonaws.dlc.e3` with the same prefix.

File: habana/pytorch/buildspec-1-10.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@ region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK pytorch
 version: &VERSION 1.10.0
 short_version: &SHORT_VERSION "1.10"
+arch_type: x86
```
Comment (tejaschumbalkar) [inline habana/pytorch/buildspec-1-10.yml:6]: Add `habana` contributor to all habana buildspec files

**Review by tejaschumbalkar — DISMISSED** [review]

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 4]
```diff
@@ -21,33 +21,76 @@ def test_dlc_major_version_label(image, region):
...
+# TODO: Apply to all images when labels are decided upon
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.integration("dlc_labels")
+@pytest.mark.model("N/A")
+def test_dlc_standard_labels(image, region):
+    customer_type_label_prefix = "e3" if test_utils.is_e3_image(image) else "sagemaker"
+    
+    # TODO: Remove skip condition when labels are added for e3 images
+    if customer_type_label_prefix == "e3":
+        pytest.skip("Labels are not currently added to e3 images")
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_dlc_labels.py:41]: Should we skip the test or confirm if there is no label attached to the image?

File: test/test_utils/__init__.py [hunk 5]
```diff
@@ -1573,3 +1601,17 @@ def is_image_available_locally(image_uri):
...
+    Return contributor name if it is present in the image URI
+
+    @param image_uri: ECR image uri
+    @return: contributor name, or ""
+    """
+    if "huggingface" in image_uri:
+        return "huggingface"
+    if "habana" in image_uri:
+        return "habana"
+    return ""
```
[the remark this answers is not in the document]
Comment (arjkesh) [reply test/test_utils/__init__.py:1617]: sure, added this in next revision

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 6]
```diff
@@ -21,33 +21,76 @@ def test_dlc_major_version_label(image, region):
...
+    for label in expected_labels:
+        assert label in actual_labels, \
+            f"Label {label} is expected in image {image}, but cannot be found. All labels on image: {actual_labels}"
 
-    test_utils.LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
+
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.integration("dlc_labels")
+@pytest.mark.model("N/A")
+def test_max_10_sagemaker_labels(image, region):
```
[the remark this answers is not in the document]
Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:86]: This is a test, and we're explicitly testing for this number. I can change the name to "test_max_sagemaker_labels" and include "10" as a variable with comments, but I don't think argument makes sense for the test itself - maybe if it was a utility function

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 7]
```diff
@@ -21,33 +21,76 @@ def test_dlc_major_version_label(image, region):
...
-    manifest_str = response["images"][0]["imageManifest"]
-    # manifest_str is a json-format string
-    manifest = json.loads(manifest_str)
-    image_metadata = json.loads(manifest["history"][0]["v1Compatibility"])
-    major_version = image_metadata["config"]["Labels"].get("dlc_major_version", None)
+    actual_labels = _get_labels_from_ecr(image, region)
 
-    assert major_version, f"{image} has no LABEL named 'dlc_major_version'. Please insert label."
+    for label in expected_labels:
+        assert label in actual_labels, \
```
[the remark this answers is not in the document]
Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:79]: sure, we can update this

File: test/dlc_tests/sanity/test_dlc_labels.py (hunk 4 above)
Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:41]: Updated the test to ensure e3 labels are not added

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 8]
```diff
@@ -175,3 +218,31 @@ class DLCMajorVersionLabelNotFound(Exception):
 
 class DLCPythonVersionNotFound(Exception):
     pass
+
+
+def _get_labels_from_ecr(image_uri, region):
```
[the remark this answers is not in the document]
Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:226]: sure, added

**Review by shantanutrip — COMMENTED** [review]
LGTM!! Lets revert the temp changes

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 9]
```diff
@@ -21,35 +19,91 @@ def test_dlc_major_version_label(image, region):
...
 
     assert major_version, f"{image} has no LABEL named 'dlc_major_version'. Please insert label."
 
     test_utils.LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
 
 
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.integration("dlc_labels")
+@pytest.mark.model("N/A")
+def test_dlc_standard_labels(image, region):
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_dlc_labels.py:33]: Just to ensure, this test would not run on the canaries, right?

Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:33]: these don't run on canaries but can be added by using the canary marker. Though they will fail for images that don't have the labels, so if added as canaries, we should ensure we have added labels to all of the images

Comment (shantanutrip) [reply test/dlc_tests/sanity/test_dlc_labels.py:33]: Yeah exactly, I wanted to confirm that we do not run them on canaries. Thanks!!

**Review by shantanutrip — APPROVED** [review]

**Review by shantanutrip — APPROVED** [review]

**Review by gradientsky — APPROVED** [review]
